### PR TITLE
Add filter support and TimeWegithedStoreRetriever in pinecone.py

### PR DESCRIPTION
**Pull request by Vasilije1990**
# Adds support for Pinecone filters when using Pinecone.from_existing_index()

This functionality extends  Pinecone.from_existing_index() function by adding filters.
The use case I was solving for was -> I need memories of a particular user, but there is no way to filter for that user ID, so I need a more robust way to do it

I also added TimeWeightedStoreRetriever as a first pass.

## Who can review?

@dev2049 can you please have a look?

### CODE REVIEW

File: langchain/vectorstores/pinecone.py [hunk 1]
```diff
@@ -37,7 +49,9 @@ def __init__(
         index: Any,
         embedding_function: Callable,
         text_key: str,
+        filter: Optional[dict] = None,
```
Comment (dev2049) [inline langchain/vectorstores/pinecone.py:52]: out of curiosity how were you imagining using this? why not just pass in the filter argument to search methods

Comment (hwchase17) [reply langchain/vectorstores/pinecone.py:52]: i would just do this by specifying **search_kwargs when converting into a retriever. or are you using for something else besides retrieval?

Comment (Vasilije1990) [reply langchain/vectorstores/pinecone.py:52]: Not really, I can remove it then

**Comment by leo-gan** [comment]
@Vasilije1990 Hi , could you, please, resolve the merging issues? After that, ping me and I push this PR for the review. Thanks!

**Comment by efriis** [comment]
Closing because the PR wouldn't line up with the current directory structure of the library (would need to be in /libs/langchain/langchain instead of /langchain). Feel free to reopen against the current head if it's still relevant!

**Pull request closed without merging**